Update smtp.markdown

gmlupatelli opened the pull request:
## Proposed change
Replaced the generic SMTP server in the Google Mail sample configuration for the right one "smtp.gmail.com", and merged the duplicate Google Mail extra steps paragraphs, clarifying a notification breaking behavior of Google less secure access option, where Google just turn it off automatically if it is not used for an undisclosed period of time.

## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [x] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: N/A
- Link to parent pull request in the Brands repository: N/A
- This PR fixes or closes issue: N/A

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

gmlupatelli committed:
Update smtp.markdown

Replaced the generic smtp server in the Google Mail sample configuration for the right one "smtp.gmail.com", and merged the duplicate Google Mail extra steps paragraphs, clarifying an annoying less secure notification behaviour, where Google just turn this access off if it is not used for an undisclosed period of time.

frenck commented on source/_integrations/smtp.markdown:
````diff
@@ -98,7 +98,9 @@ notify:
 ```
 
-Keep in mind that Google has some extra layers of protection which need special attention (Hint: 'Less secure apps'). If you have 2-step verification enabled on your Google account, you'll need to use [an application-specific password](https://support.google.com/mail/answer/185833?hl=en).
+Keep in mind that Google has some extra layers of protection which need special attention. By default, the usage by external applications is limited so you will need to visit the [less secure apps](https://www.google.com/settings/security/lesssecureapps) page and enable it to be able to send e-mails, and be aware that Google will periodically turn it off if it is not used (no e-mail is sent).
+
+To avoid having your e-mail notifications broken due to the less secure apps behaviour, it is recommended that you enable 2-step verification on your Google account, and use [an application-specific password](https://support.google.com/mail/answer/185833?hl=en) in your notification configuration.
````
This is very specific content that only applies to a specific email provider. Lets move this down in the documentation in a separate chapter, or even completely remove it.

gmlupatelli replied:
I agree........I just didn't do that as I thought it was up for some reason. I will keep in the top a generic SMTP configuration example and create a chapter at the end of the page for specific e-mail provider configuration examples.

frenck commented on source/_integrations/smtp.markdown:
```diff
@@ -85,7 +85,7 @@ A sample configuration entry for Google Mail.
   - name: "NOTIFIER_NAME"
     platform: smtp
-    server: "smtp.example.com"
+    server: "smtp.gmail.com"
```
Please use `example.com` is it designed/documented to use for examples by RFC2606

gmlupatelli replied:
I will do that and reword the text that comes before the example to make it clear that this is not a google mail example but a generic SMTP example.

gmlupatelli committed:
Updated smtp page according to the feedback

Updated the generic Example configuration.yaml entry including the SMTP server, user name, and password fields. 
Separate the configuration segment from the usage segment by including a USAGE chapter (got the inspiration from the Facebook notification).
Moved the provider-specific configuration to the bottom of the page in a new chapter called Specific E-Mail Provider Configuration.

frenck commented on source/_integrations/smtp.markdown:
```diff
@@ -21,9 +21,13 @@ To enable notification by e-mail in your installation, add the following to your
   - name: "NOTIFIER_NAME"
     platform: smtp
+    server: "smtp.example.com"
+    username: "YOUR_USER_NAME"
+    password: "YOUR_PASSWORD"
```
We try to keep all configuration samples minimal. Thus, no optional requirement in the default sample.

gmlupatelli replied:
I was divided in this one, on one hand, we have the pattern of not including optional configurations in the configuration example, and on the other, we have that the most common use case for SMTP integrations is to use an external SMTP provider that will usually require a username and password.

Given that we are including provider-specific configuration in the end I will not strongly argue about keeping those optional Configuration variables, so I am good about removing them if I could not convince you @frenck :)

frenck replied:
There is no discussion on this part, this is following our documentation standards. Please remove those optional parameters. Thanks 👍

gmlupatelli replied:
Consider it done! I will revert to the original example, below, as the server is also an optional variable.
`# Example configuration.yaml entry
notify:
  - name: "NOTIFIER_NAME"
    platform: smtp
    sender: "YOUR_SENDER"
    recipient: "YOUR_RECIPIENT"`

frenck commented on source/_integrations/smtp.markdown:
````diff
@@ -21,9 +21,13 @@ To enable notification by e-mail in your installation, add the following to your
     sender: "YOUR_SENDER"
     recipient: "YOUR_RECIPIENT"
 ```
+Check your e-mail provider configuration or help pages to get the correct SMTP settings.
````
```suggestion

Check your e-mail provider configuration or help pages to get the correct SMTP settings.
```

frenck commented on source/_integrations/smtp.markdown:
````diff
@@ -192,8 +174,33 @@ burglar:
+      - "RECIPIENT_N@example.com"
+    sender_name: "SENDER_NAME"
+```
+Keep in mind that Google has some extra layers of protection which need special attention. By default, the usage by external applications is limited so you will need to visit the [less secure apps](https://www.google.com/settings/security/lesssecureapps) page and enable it to be able to send e-mails, and be aware that Google will periodically turn it off if it is not used (no e-mail is sent).
````
```suggestion

Keep in mind that Google has some extra layers of protection that need special attention. By default, the usage by external applications is limited so you will need to visit the [less secure apps](https://www.google.com/settings/security/lesssecureapps) page and enable it to be able to send e-mails. Be aware that Google will periodically turn it off if it is not used (no e-mail is sent).
```

frenck commented on source/_integrations/smtp.markdown:
````diff
@@ -192,8 +174,33 @@ burglar:
+```
+Keep in mind that Google has some extra layers of protection which need special attention. By default, the usage by external applications is limited so you will need to visit the [less secure apps](https://www.google.com/settings/security/lesssecureapps) page and enable it to be able to send e-mails, and be aware that Google will periodically turn it off if it is not used (no e-mail is sent).
+
+To avoid having your e-mail notifications broken due to the less secure apps behaviour, it is recommended that you enable 2-step verification on your Google account, and use [an application-specific password](https://support.google.com/mail/answer/185833?hl=en) in your notification configuration.
````
```suggestion
To avoid having your e-mail notifications broken due to the less secure app's behavior, it is recommended that you enable 2-step verification on your Google account, and use [an application-specific password](https://support.google.com/mail/answer/185833?hl=en) in your notification configuration.
```

frenck commented on source/_integrations/smtp.markdown:
```diff
@@ -192,8 +174,33 @@ burglar:
+To learn more about how to use notifications in your automations, please see the [getting started with automation page](/getting-started/automation/).
 
-A combination that will work properly is port 587 and STARTTLS. It's recommended to enable STARTTLS, if possible.
+## Specific E-Mail Provider Configuration
+Check below some configurations examples for specific e-mail providers. 
+If you are in doubt about the SMTP settings required, check your e-mail provider configuration or help pages for more information about its specific SMTP configuration. 
```
```suggestion
## Specific E-Mail Provider Configuration

Check below some configurations examples for specific e-mail providers. 
If you are in doubt about the SMTP settings required, check your e-mail provider configuration or help pages for more information about its specific SMTP configuration. 
```

gmlupatelli committed:
Update source/_integrations/smtp.markdown

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

gmlupatelli committed:
Update source/_integrations/smtp.markdown

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

gmlupatelli committed:
Update source/_integrations/smtp.markdown

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

gmlupatelli committed:
Update source/_integrations/smtp.markdown

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

gmlupatelli committed:
Revert the initial yaml example to the original one without any optional variable

Revert the initial yaml example to the original one without any optional variable

frenck reviewed (approved):
Thanks, @gmlupatelli 👍
